src/libraries/System.Security.Cryptography/tests/X509Certificates/FindTests.cs

```diff
@@ -958,6 +958,233 @@ public static void TestFindByKeyUsage_NoMatch(object matchCriteria)
+                Assert.Empty(coll.FindByThumbprint(hashAlgorithm, thumbprintHexTrailing.AsSpan()));
+                Assert.Empty(coll.FindByThumbprint(hashAlgorithm, thumbprintTrailing));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ThumbprintDigestAlgorithms))]
+        public static void FindByThumbprint_ByAlgorithm_PrependedData(HashAlgorithmName hashAlgorithm)
+        {
+            using (X509Certificate2 cert = new(TestData.CertWithTemplateData))
+            using (X509Certificate2 unrelated = new(TestData.CertWithPolicies))
+            {
+                byte[] thumbprint = CryptographicOperations.HashData(hashAlgorithm, cert.RawDataMemory.Span);
+
+                ReadOnlySpan<byte> thumbprintPrepended = [0xF0, 0x0D, ..thumbprint];
+                string thumbprintHexPrepended = Convert.ToHexString(thumbprintPrepended);
+
+                X509Certificate2Collection coll = [cert, unrelated];
+                Assert.Empty(coll.FindByThumbprint(hashAlgorithm, thumbprintHexPrepended));
+                Assert.Empty(coll.FindByThumbprint(hashAlgorithm, thumbprintHexPrepended.AsSpan()));
+                Assert.Empty(coll.FindByThumbprint(hashAlgorithm, thumbprintPrepended));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ThumbprintDigestAlgorithms))]
+        public static void FindByThumbprint_ByAlgorithm_Truncated(HashAlgorithmName hashAlgorithm)
+        {
+            using (X509Certificate2 cert = new(TestData.CertWithTemplateData))
+            using (X509Certificate2 unrelated = new(TestData.CertWithPolicies))
+            {
+                byte[] thumbprint = CryptographicOperations.HashData(hashAlgorithm, cert.RawDataMemory.Span);
+
+                ReadOnlySpan<byte> thumbprintTruncated = thumbprint.AsSpan(0..^1);
+                string thumbprintHexTruncated = Convert.ToHexString(thumbprintTruncated);
+
+                X509Certificate2Collection coll = [cert, unrelated];
+                Assert.Empty(coll.FindByThumbprint(hashAlgorithm, thumbprintHexTruncated));
+                Assert.Empty(coll.FindByThumbprint(hashAlgorithm, thumbprintHexTruncated.AsSpan()));
+                Assert.Empty(coll.FindByThumbprint(hashAlgorithm, thumbprintTruncated));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ThumbprintDigestAlgorithms))]
+        public static void FindByThumbprint_ByAlgorithm_CaseInsensitive(HashAlgorithmName hashAlgorithm)
+        {
+            using (X509Certificate2 cert = new(TestData.CertWithTemplateData))
+            using (X509Certificate2 unrelated = new(TestData.CertWithPolicies))
+            {
+                byte[] thumbprint = CryptographicOperations.HashData(hashAlgorithm, cert.RawDataMemory.Span);
+
+                string thumbprintHexUpper = Convert.ToHexString(thumbprint);
+                string thumbprintHexLower = Convert.ToHexStringLower(thumbprint);
```

Comment:
Since the thumbprint isn't visible here, is it worth asserting that there's at least one non-numeric character?  != on the two strings works, as does IndexOfAnyExcept('0', '1', ..., '9')

Reply:
I guess so, it would be super funny if the thumbprint worked out to be only numeric digits.